classes/class-wc-connect-service-settings-store.php

```diff
@@ -110,6 +110,40 @@ public function can_user_manage_payment_methods() {
 				( is_a( $master_user, 'WP_User' ) && $current_user->ID === $master_user->ID );
 		}
 
+		/**
+		 * Gets the store's location settings.
+		 *
+		 * @return array
+		 */
+		public function get_endshipper_address() {
+			$email       = '';
+			$master_user = WC_Connect_Jetpack::get_master_user();
+
+			if ( is_a( $master_user, 'WP_User' ) ) {
+
+				$connected_data = WC_Connect_Jetpack::get_connected_user_data( $master_user->ID );
+
+				if ( is_array( $connected_data ) && ! empty( $connected_data['email'] ) ) {
```

Comment:
Should we exit if we couldn't retrieve an email?

```suggestion
				if ( empty( $connected_data['email'] ) ) {
					throw new SOME_EXCEPTION;
```

Reply:
It seems there is no exception catch around this code yet. So if we throw any exception, it will show a fatal error instead.

What I can suggest probably, we add a try/catch around [this code](https://github.com/Automattic/woocommerce-services/blob/4b160bcfb1917b92450a86893cbc292efd374af7/classes/class-wc-connect-shipping-label.php#L554-L577). Here is how we can implement it : 

```
public function meta_box( $post, $args ) {
			try {
				$connect_order_presenter = new WC_Connect_Order_Presenter();
				$order                   = WC_Connect_Compatibility::instance()->init_theorder_object( $post );
				...
				...
				...

				do_action( 'enqueue_wc_connect_script', 'wc-connect-create-shipping-label', $payload );
			} catch ( Exception $e ) {
				?>
				<strong><?php echo sprintf( esc_html__( 'Error: %1$s', 'wocommerce-services' ), $e->getMessage() ); ?></strong>
				<?php
			}
		}
```
WDYT?